vm/devices/storage/disklayer_ram/src/lib.rs

```diff
@@ -552,4 +611,234 @@ mod tests {
+            0,
+        )
+        .await
+        .unwrap();
+
+        let mut read_data = vec![0u8; SECTOR_4K as usize];
+        guest_mem.read_at(0, &mut read_data).unwrap();
+        assert_eq!(read_data, test_data);
+    }
+
+    #[async_test]
+    async fn test_4k_sectors_multiple() {
+        // Test multiple 4K sectors using the convenience function
+        const SIZE: u64 = 16 * 4096; // 64KB = 16 sectors
+
+        let disk = super::ram_disk_with_sector_size(SIZE, false, SECTOR_4K).unwrap();
+        let guest_mem = GuestMemory::allocate(SECTOR_4K as usize);
+
+        // Write different patterns to different sectors
+        for sector in 0..4 {
+            let pattern = (sector + 1) as u8;
+            let sector_data = vec![pattern; SECTOR_4K as usize];
+            guest_mem.write_at(0, &sector_data).unwrap();
+
+            disk.write_vectored(
+                &OwnedRequestBuffers::linear(0, SECTOR_4K as usize, false).buffer(&guest_mem),
+                sector,
+                false,
+            )
+            .await
+            .unwrap();
+        }
+
+        // Read back and verify each sector
+        for sector in 0..4 {
+            guest_mem.fill_at(0, 0, SECTOR_4K as usize).unwrap();
+            disk.read_vectored(
+                &OwnedRequestBuffers::linear(0, SECTOR_4K as usize, true).buffer(&guest_mem),
+                sector,
+            )
+            .await
+            .unwrap();
+
+            let mut read_data = vec![0u8; SECTOR_4K as usize];
+            guest_mem.read_at(0, &mut read_data).unwrap();
+            let expected_pattern = (sector + 1) as u8;
+            assert!(read_data.iter().all(|&b| b == expected_pattern));
+        }
+    }
+
+    #[async_test]
+    async fn test_4k_ram_disk_helper() {
+        // Test the convenience function with 4K sectors
+        const SIZE: u64 = 8 * 4096; // 32KB
+
+        let disk = super::ram_disk_with_sector_size(SIZE, false, SECTOR_4K).unwrap();
+        let guest_mem = GuestMemory::allocate(SECTOR_4K as usize);
+
+        // Write a test pattern
+        let test_data: Vec<u8> = (0..SECTOR_4K).map(|i| (i / 16) as u8).collect();
+        guest_mem.write_at(0, &test_data).unwrap();
+
+        disk.write_vectored(
+            &OwnedRequestBuffers::linear(0, SECTOR_4K as usize, false).buffer(&guest_mem),
+            1, // Write to sector 1
+            false,
+        )
+        .await
+        .unwrap();
+
+        // Read back
+        guest_mem.fill_at(0, 0, SECTOR_4K as usize).unwrap();
+        disk.read_vectored(
+            &OwnedRequestBuffers::linear(0, SECTOR_4K as usize, true).buffer(&guest_mem),
+            1,
+        )
+        .await
+        .unwrap();
+
+        let mut read_data = vec![0u8; SECTOR_4K as usize];
+        guest_mem.read_at(0, &mut read_data).unwrap();
+        assert_eq!(read_data, test_data);
+    }
+
+    #[async_test]
+    async fn test_mixed_sector_sizes() {
```

Comment:
This test seems pointless unless you combine the layers into a single disk and make sure the visible sector size matches what you'd expect.